src/compiler/interface2.jl

```diff
@@ -35,13 +35,17 @@ end
 end
 
 @generated function (j::Pullback{T})(Δ) where T
-  ignore_sig(T) && return :nothing
+  ignore_sig(T) && return :(DoesNotExist())
   g = try _lookup_grad(T)
   catch e
     rethrow(CompileError(T,e))
   end
   if g == nothing
-    Δ == Nothing && return :nothing
+    Δ isa AbstractZero && return :(DoesNotExist())
+    if Δ == Nothing
+        Core.println("'nothing' for zero gradients is deprecated. Use Zero() from ChainRules.")
```

Comment:
This probably can't stay here.
It probably needs to move to be returned  as part of the next line.